packages/tempus-client_v3/src/services/NotificationService.spec.ts

```diff
@@ -0,0 +1,308 @@
+          link: 'someLink',
+          linkText: 'someLinkText',
+          timestamp: 1649031780000,
+          title: 'Notify T',
+        },
+      ]);
+    });
+
+    it('does not store a notification Wallet', () => {
+      notificationService.notify('fantom', 'Wallet', 'Notify W', 'unimportant');
+
+      expect(mockSet).not.toHaveBeenCalled();
+    });
+
+    it('does not store a notification Service', () => {
+      notificationService.notify('ethereum', 'Service', 'Notify S', 'some value');
+
+      expect(mockSet).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('getNextItem()', () => {
+    beforeEach(() => {
+      mockGet.mockReturnValue([]);
+      notificationService = new NotificationService(mockStorageService);
+    });
+
+    it('emits notifications in the same order as they are created', () => {
+      notificationService.getNextItem().subscribe(mockListener);
+
+      notificationService.notify(
+        'fantom',
+        'Transaction',
+        'Notification 1',
+        'notification content',
+        'someLink',
+        'someLinkText',
+      );
+
+      notificationService.warn(
+        'fantom',
+        'Transaction',
+        'Notification 2',
+        'warning content',
+        'someLink',
+        'someLinkText',
+      );
+
+      expect(mockListener).toHaveBeenNthCalledWith(1, {
+        category: 'Transaction',
+        chain: 'fantom',
+        content: 'notification content',
+        dismissed: false,
+        id: 'abc000',
+        level: 'info',
+        link: 'someLink',
+        linkText: 'someLinkText',
+        timestamp: 1649031780000,
+        title: 'Notification 1',
+      });
+
+      expect(mockListener).toHaveBeenNthCalledWith(2, {
+        category: 'Transaction',
+        chain: 'fantom',
+        content: 'warning content',
+        dismissed: false,
+        id: 'abc000',
+        level: 'warning',
+        link: 'someLink',
+        linkText: 'someLinkText',
+        timestamp: 1649031780000,
+        title: 'Notification 2',
+      });
+    });
+  });
+
+  describe('getLastItems()', () => {
+    beforeEach(() => {
+      mockGet.mockReturnValue(mockStoredNotifications);
+      notificationService = new NotificationService(mockStorageService);
+    });
+
+    afterAll(() => {
+      mockGet.mockReturnValue([]);
+    });
+
+    it('emits last stored notifications', () => {
+      expect(mockListener).not.toHaveBeenCalled();
+
+      notificationService.getLastItems().subscribe(mockListener);
+
+      expect(mockListener).not.toHaveBeenCalled();
+
+      act(() => {
+        jest.advanceTimersByTime(100);
+      });
+
+      expect(mockDelete).toHaveBeenCalledTimes(1);
+
+      expect(mockListener).toHaveBeenNthCalledWith(1, {
+        category: 'Transaction',
+        chain: 'ethereum',
+        content: 'content 3',
+        dismissed: false,
+        id: 'stored003',
+        level: 'warning',
+        timestamp: 1652569200010,
+        title: 'StoredNotification 3',
+      });
```

Comment:
(to continue the previous comment)

... and use the constants here:
```suggestion
      expect(mockListener).toHaveBeenNthCalledWith(1, notification3);
```

Reply:
refactored